src/renderer/services/dex.js

```diff
@@ -788,59 +809,84 @@ export default {
       try {
         let neoToSend = 0;
         let gasToSend = 0;
+        let holding = null;
 
         if (assetId === assets.NEO) {
-          const neoHolding = neo.getHolding(assets.NEO);
+          holding = neo.getHolding(assets.NEO);
           neoToSend = quantity;
-          if (neoToSend > neoHolding.balance) {
+          if (neoToSend > holding.balance) {
             reject('Insufficient NEO.');
             return;
           }
-        }
-
-        if (assetId === assets.GAS) {
-          const gasHolding = neo.getHolding(assets.GAS);
+        } else if (assetId === assets.GAS) {
+          holding = neo.getHolding(assets.GAS);
           gasToSend = quantity;
-          if (gasToSend > gasHolding.balance) {
+          if (gasToSend > holding.balance) {
             reject('Insufficient GAS.');
             return;
           }
+        } else {
+          holding = neo.getHolding(assetId);
+          if (holding.balance.isLessThan(quantity)) {
+            reject(`Insufficient ${holding != null ? holding.symbol : assetId}.`);
+            return;
+          }
         }
 
-        this.executeContractTransaction('deposit',
-          [
-            u.reverseHex(assetId),
-            u.num2fixed8(quantity),
-          ], neoToSend, gasToSend)
-          .then((res) => {
-            if (res.success) {
-              alerts.success('Deposit relayed, waiting for confirmation...');
-              neo.monitorTransactionConfirmation(res.tx, true)
-                .then(() => {
-                  const inMemoryHolding = _.get(store.state.nep5Balances, assetId);
-                  if (inMemoryHolding) {
-                    inMemoryHolding.balance = null;
-                  }
+        if (holding.canPull !== false) {
+          this.executeContractTransaction('deposit',
+            [
+              u.reverseHex(assetId),
+              u.num2fixed8(quantity),
+            ], neoToSend, gasToSend)
+            .then((res) => {
+              if (res.success) {
+                alerts.success('Deposit relayed, waiting for confirmation...');
+                neo.monitorTransactionConfirmation(res.tx, true)
+                  .then(() => {
+                    const inMemoryHolding = _.get(store.state.nep5Balances, assetId);
+                    if (inMemoryHolding) {
+                      inMemoryHolding.balance = null;
+                    }
 
-                  resolve(res.tx);
-                })
-                .catch((e) => {
-                  reject(e);
-                });
-            } else {
-              reject('Transaction rejected');
-            }
+                    resolve(res.tx);
+                  })
+                  .catch((e) => {
+                    reject(`Deposit Failed. ${e.message}`);
```

Comment:
Same here with i18n; there are a few other spots.

Reply:
I don't believe any of the error messages in js files are localized, this should be a separate ticket. I don't see an example of how to import the localization into a service.